source/common/tracing/http_tracer_impl.cc

```diff
@@ -172,6 +172,10 @@ void HttpTracerUtility::finalizeDownstreamSpan(Span& span, const Http::HeaderMap
     span.setTag(Tracing::Tags::get().UserAgent, valueOrDefault(request_headers->UserAgent(), "-"));
     span.setTag(Tracing::Tags::get().HttpProtocol,
                 AccessLog::AccessLogFormatUtils::protocolToString(stream_info.protocol()));
+    const auto ip = stream_info.downstreamDirectRemoteAddress()->ip();
+    if (ip != nullptr) {
```

Comment:
I don't think you need this check here. The documentation says that this will be valid if the type of the Address is IP (i.e. not a unix domain socket for example or a pipe)
I'd change it to an `assert`. I'd also assert that `downstreamDirectRemoteAddress()` is not `nullptr` (the documentation says it will never be `nullptr`)

Reply:
Done. Let me know if I misinterpreted your comment.